src/crewai/agents/executor.py

```diff
@@ -185,6 +197,52 @@ def _iter_next_step(
             yield AgentStep(action=output, observation=observation)
             return
 
+        except BadRequestError as e:
```

Comment:
_Suggestion_: If the error raised is a type-specific error, maybe is worth it to create a Custom Exception and handle the specific cases when we have the context length exceeded this way you can have something like

`except ContextLengthExceeded as exc`